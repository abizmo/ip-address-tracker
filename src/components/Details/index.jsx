import { useState } from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  width: 75%;
`;

function Details() {
  const [ip] = useState('192.212.174.101');
  const [location] = useState('Brooklyn, NY 10001');
  const [timezone] = useState('UTC');
  const [offset] = useState('-05:00');
  const [isp] = useState('SpaceX Starlink');

  return (
    <Container>
      <div>
        <h2>IP Address</h2>
        <p>{ip}</p>
      </div>
      <div>
        <h2>Location</h2>
        <p>{location}</p>
      </div>
      <div>
        <h2>Timezone</h2>
        <p>
          {timezone}
          {' '}
          {offset}
        </p>
      </div>
      <div>
        <h2>ISP</h2>
        <p>{isp}</p>
      </div>
    </Container>
  );
}

export default Details;
