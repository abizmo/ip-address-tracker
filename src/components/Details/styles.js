import styled from 'styled-components';

export const Card = styled.div`
  align-items: center;
  background-color: var(--clr-100);
  border-radius: 1rem;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  padding: 1.5rem;
  position: absolute;
  width: calc(100% - 48px);
`;

export const Label = styled.h2`
  color: var(--clr-400);
  font-size: var(--fs-100);
  font-weight: 700;
  letter-spacing: 0.16em;
  margin-bottom: .5rem;
  text-align: center;
  text-transform: uppercase;
`;

export const Detail = styled.p`
  font-size: var(--fs-500);
  font-weight: 500;
  letter-spacing: -0.01em;
  text-align: center;
`;
