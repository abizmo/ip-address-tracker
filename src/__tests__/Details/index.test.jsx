import { shallow } from 'enzyme';
import Details from '../../components/Details';

describe('Details test', () => {
  test('should render and match', () => {
    const wrapper = shallow(<Details />);
    const titles = wrapper.find('h2').children();

    expect(wrapper).toMatchSnapshot();
    expect(titles).toHaveLength(4);
    expect(titles.first().text()).toBe('IP Address');
  });
});
