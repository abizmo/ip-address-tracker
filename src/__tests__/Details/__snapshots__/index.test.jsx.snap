// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test should render and match 1`] = `
<styled.div>
  <div>
    <h2>
      IP Address
    </h2>
    <p>
      192.212.174.101
    </p>
  </div>
  <div>
    <h2>
      Location
    </h2>
    <p>
      Brooklyn, NY 10001
    </p>
  </div>
  <div>
    <h2>
      Timezone
    </h2>
    <p>
      UTC
       
      -05:00
    </p>
  </div>
  <div>
    <h2>
      ISP
    </h2>
    <p>
      SpaceX Starlink
    </p>
  </div>
</styled.div>
`;
